packages/core/src/context/ContextEnginePipeline.ts

```
@@ -0,0 +1,738 @@
+  ): Promise<{
+    vectorResults: SearchResult[];
+    memoryResults: MemorySearchResult[];
+    warnings: string[];
+    servicesUsed: string[];
+  }> {
+    const warnings: string[] = [];
+    const servicesUsed: string[] = [];
+
+    switch (options.strategy) {
+      case ContextStrategy.VECTOR_ONLY:
+        const vectorResults = await this.retrieveRelevantContext(query, options.vectorSearchLimit, abortSignal);
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)